lib/puppet/functions/length.rb

```diff
@@ -11,4 +11,5 @@ def length(value)
     end
     result
   end
+  Puppet.deprecation_warning('The length function is deprecated. This function is now shipped with Puppet in versions 5.5.0 and later. For more information please see Puppet 5.5.0 Release Notes.')
```

Comment:
These functions should only be deprecated for users using puppet >=5.5.0, not for users running puppet <5.5.0. So it shouldn't print the deprecation warnings on puppet <5.5.0.

Users running >=5.5.0 will not be able to call these functions without specially trying, so I don't think printing a deprecating warning for >=5.5.0 will actually be helpful.

So we probably don't need to print them at all, and thus don't need to test for them.